desc/equilibrium/coords.py

```diff
@@ -138,19 +138,19 @@ def map_coordinates(  # noqa: C901
 
     # do surface average to get iota once
     if "iota" in profiles and profiles["iota"] is None:
-        profiles["iota"] = eq.get_profile("iota", params=params)
+        profiles["iota"] = eq.get_profile(["iota", "iota_r"], params=params)
         params["i_l"] = profiles["iota"].params
 
     @functools.partial(jit, static_argnums=1)
     def compute(y, basis):
         grid = Grid(y, sort=False, jitable=True)
         data = {}
         if "iota" in deps:
-            data["iota"] = profiles["iota"](grid, params=params["i_l"])
+            data["iota"] = profiles["iota"].compute(grid, params=params["i_l"])
```

Comment:
in the ripple branch, i noticed you get issues when map_coordinates is used inside jitted objective.compute. The switch to `.compute` here and other changes are necessary to fix the issue

Reply:
I'm surprised this made any difference, if you supply a grid then the call method is literally the same as compute just with the order swapped?

Reply:
Ah right, it doesn't matter on this line, but it does on line 246 below